classes/Tools.php

```diff
@@ -734,42 +738,95 @@ public static function getCldr(Context $context = null, $language_code = null)
+     *
+     * @throws LocalizationException
      */
     public static function displayPrice($price, $currency = null, $no_utf8 = false, Context $context = null)
     {
+        @trigger_error(
+            'Tools::displayPrice() is deprecated since version 1.7.5.0. '
+            . 'Use PrestaShop\PrestaShop\Core\Localization\Locale::formatPrice() instead.',
+            E_USER_DEPRECATED
+        );
+
         if (!is_numeric($price)) {
             return $price;
         }
-        if (!$context) {
-            $context = Context::getContext();
+
+        $context = $context ?: Context::getContext();
+        $currency = $currency ?: $context->currency;
+
+        if (is_int($currency)) {
+            $currency = Currency::getCurrencyInstance($currency);
         }
-        if ($currency === null) {
-            $currency = $context->currency;
-        } elseif (is_int($currency)) {
-            $currency = Currency::getCurrencyInstance((int) $currency);
+
+        $locale = static::getContextLocale($context);
+        $currencyCode = is_array($currency) ? $currency['iso_code'] : $currency->iso_code;
+
+        return $locale->formatPrice($price, $currencyCode);
+    }
+
+    protected static function getContextLocale(Context $context)
+    {
+        $locale = $context->currentLocale;
+        if (null !== $locale) {
+            return $locale;
+        }
+
+        $container = isset($context->controller) ? $context->controller->getContainer() : null;
+        if (null === $container) {
+            $container = SymfonyContainer::getInstance();
         }
 
-        $cldr = self::getCldr($context);
+        /** @var LocaleRepository $localeRepository */
+        $localeRepository = $container->get(self::SERVICE_LOCALE_REPOSITORY);
+        $locale = $localeRepository->getLocale((string) $context->language->locale);
 
-        return $cldr->getPrice($price, is_array($currency) ? $currency['iso_code'] : $currency->iso_code);
+        return $locale;
     }
 
     /**
-     * Return a number well formatted
+     * Returns a well formatted number.
+     *
+     * @deprecated Since 1.7.5.0. Please use Locale::formatNumber() instead
+     * @see PrestaShop\PrestaShop\Core\Localization\Locale
+     *
+     * @param float $number
+     *                      The number to format
+     * @param null $currency
+     *                       not used anymore
+     *
+     * @return string
+     *                The formatted number
      *
-     * @param float $number A number
-     * @param nullable $currency / not used anymaore
+     * @throws Exception
+     * @throws LocalizationException
      */
     public static function displayNumber($number, $currency = null)
     {
-        $cldr = self::getCldr(Context::getContext());
+        @trigger_error(
+            'Tools::displayNumber() is deprecated since version 1.7.5.0. '
+            . 'Use PrestaShop\PrestaShop\Core\Localization\Locale::formatNumber() instead.',
```

Comment:
use Locale::class instead in case we move or do something with the classe in the future.
And deprecated since 1.7.6.0